tools/v3benchmark/main.go

```diff
@@ -50,53 +55,32 @@ func main() {
 		rangeEnd = []byte(flag.Args()[2])
 	}
 
-	results := make(chan *result, n)
-	bar := pb.New(n)
+	results = make(chan *result, n)
+	bar = pb.New(n)
 	bar.Format("Bom !")
 	bar.Start()
+
 	start := time.Now()
-	defer func() {
-		bar.Finish()
-		printReport(n, results, time.Now().Sub(start))
-	}()
 
-	var wg sync.WaitGroup
 	wg.Add(c)
-	jobs := make(chan struct{}, n)
-	for i := 0; i < c; i++ {
-		go func() {
-			defer wg.Done()
-
-			conn, err := grpc.Dial(url)
-			if err != nil {
-				fmt.Errorf("dial error: %v", err)
-				os.Exit(1)
-			}
-			etcd := etcdserverpb.NewEtcdClient(conn)
-			req := &etcdserverpb.RangeRequest{Key: key, RangeEnd: rangeEnd}
-
-			for _ = range jobs {
-				st := time.Now()
-				resp, err := etcd.Range(context.Background(), req)
+	requests := make(chan struct{}, n)
+	conn, err := grpc.Dial(url)
+	if err != nil {
+		fmt.Errorf("dial error: %v", err)
+		os.Exit(1)
+	}
```

Comment:
I will just stick to what gRPC official does today. I would like to explore more in the future.